util/unixfs.go

```diff
@@ -41,3 +44,21 @@ func ImportFile(dserv ipld.DAGService, fi io.Reader) (ipld.Node, error) {
 
 	return balanced.Layout(db)
 }
+
+func TryExtractFSNode(nd ipld.Node) (*unixfs.FSNode, error) {
+	switch nd := nd.(type) {
+	case *merkledag.ProtoNode:
+		n, err := unixfs.FSNodeFromBytes(nd.Data())
+		if err != nil {
+			return nil, err
+		}
+		if n.Type() == unixfs.TSymlink {
+			return nil, fmt.Errorf("symlinks not supported")
+		}
+		return n, nil // success!
+	case *merkledag.RawNode:
```

Comment:
We should treat RawNodes as files, I think this will error out and FindCIDType will return unknown.